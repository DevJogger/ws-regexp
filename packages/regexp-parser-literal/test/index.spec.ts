import { parsePattern, parseFlags, parseRegExp } from '../index';

test(`parsePattern`, () =>
{

	let actual = parsePattern(/麻#_@_#痹/.source);
	let expected = `麻#_@_#痹`;

	expect(actual.raw).toStrictEqual(expected);
	//expect(actual).toBeInstanceOf(Date);
	expect(actual).toMatchSnapshot();

});

test(`parseFlags`, () =>
{

	let actual = parseFlags('ygmisu');

	expect(actual).toMatchSnapshot();

});

test(`parseRegExp`, () =>
{

	let actual = parseRegExp(/麻#_@_#痹/u.toString());

	expect(actual).toMatchSnapshot();

});
