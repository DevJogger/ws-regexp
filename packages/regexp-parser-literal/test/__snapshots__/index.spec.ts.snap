// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parseFlags 1`] = `
{
  "dotAll": true,
  "end": 6,
  "global": true,
  "ignoreCase": true,
  "multiline": true,
  "parent": null,
  "raw": "ygmisu",
  "start": 0,
  "sticky": true,
  "type": "Flags",
  "unicode": true,
}
`;

exports[`parsePattern 1`] = `
{
  "elements": [
    {
      "end": 1,
      "parent": [Circular],
      "raw": "麻",
      "start": 0,
      "type": "Character",
      "value": 40635,
    },
    {
      "end": 2,
      "parent": [Circular],
      "raw": "#",
      "start": 1,
      "type": "Character",
      "value": 35,
    },
    {
      "end": 3,
      "parent": [Circular],
      "raw": "_",
      "start": 2,
      "type": "Character",
      "value": 95,
    },
    {
      "end": 4,
      "parent": [Circular],
      "raw": "@",
      "start": 3,
      "type": "Character",
      "value": 64,
    },
    {
      "end": 5,
      "parent": [Circular],
      "raw": "_",
      "start": 4,
      "type": "Character",
      "value": 95,
    },
    {
      "end": 6,
      "parent": [Circular],
      "raw": "#",
      "start": 5,
      "type": "Character",
      "value": 35,
    },
    {
      "end": 7,
      "parent": [Circular],
      "raw": "痹",
      "start": 6,
      "type": "Character",
      "value": 30201,
    },
  ],
  "end": 7,
  "parent": null,
  "raw": "麻#_@_#痹",
  "start": 0,
  "type": "Pattern",
}
`;

exports[`parseRegExp 1`] = `
{
  "end": 10,
  "flags": {
    "dotAll": false,
    "end": 10,
    "global": false,
    "ignoreCase": false,
    "multiline": false,
    "parent": [Circular],
    "raw": "u",
    "start": 9,
    "sticky": false,
    "type": "Flags",
    "unicode": true,
  },
  "parent": null,
  "pattern": {
    "elements": [
      {
        "end": 2,
        "parent": [Circular],
        "raw": "麻",
        "start": 1,
        "type": "Character",
        "value": 40635,
      },
      {
        "end": 3,
        "parent": [Circular],
        "raw": "#",
        "start": 2,
        "type": "Character",
        "value": 35,
      },
      {
        "end": 4,
        "parent": [Circular],
        "raw": "_",
        "start": 3,
        "type": "Character",
        "value": 95,
      },
      {
        "end": 5,
        "parent": [Circular],
        "raw": "@",
        "start": 4,
        "type": "Character",
        "value": 64,
      },
      {
        "end": 6,
        "parent": [Circular],
        "raw": "_",
        "start": 5,
        "type": "Character",
        "value": 95,
      },
      {
        "end": 7,
        "parent": [Circular],
        "raw": "#",
        "start": 6,
        "type": "Character",
        "value": 35,
      },
      {
        "end": 8,
        "parent": [Circular],
        "raw": "痹",
        "start": 7,
        "type": "Character",
        "value": 30201,
      },
    ],
    "end": 8,
    "parent": [Circular],
    "raw": "麻#_@_#痹",
    "start": 1,
    "type": "Pattern",
  },
  "raw": "/麻#_@_#痹/u",
  "start": 0,
  "type": "RegExpLiteral",
}
`;
